floop-data-export.py

```diff
@@ -21,34 +26,41 @@
 
 db = firestore.client()
 
+
 # pull conversation documents where 'Comment_Preview' field isn't certain
-#   text, and pulling a limit of 10k.
+#   text, and pulling a limit of 1k.
 conversations = db.collection(
     'Databases/Dev_Database/Conversations').where('Comment_Preview', 'not-in', [
-        'What is your goal for this year?', 'Audio Comment', 'Freeform', 'Freeform Comment', '', ' ']).limit(10000).get()
+        'What is your goal for this year?', 'Audio Comment', 'Freeform', 'Freeform Comment', '', ' ']).limit(n).get()
 
 if __name__ == '__main__':
 
     # Iniialize empty list, "cList"
     cList = []
-# For each item in Conversations, ea entry has a Messages collection
+    
+# For each item in Conversations, each entry has a Messages collection
 # all documents will be ordered by 'Date_Submitted' and only the first comment will be pulled.
-# For each document in the pulled list(currently limited to 1 per collection),
 #    get value of "Text" field and add to cList
 
     for convo in conversations:
+
+        # Creating dictionary for mapping Sender ID to Teacher or Student
+        t_s_mapper = convo.to_dict()['Participant_IDs']
+
         convo_entries = convo.reference.collection(
-            'Messages').order_by(u'Date_Submitted').limit(1).get()
+            'Messages').order_by(u'Date_Submitted').get()
 
         for entry in convo_entries:
-            cList.append(entry.get("Text").strip())
+            # Checking for non-empty strings only
+            if entry.get("Text").strip()!='':
+
+                cList.append({
+                    'Text': entry.get("Text").strip(),
+                    'uid': t_s_mapper[entry.get("Sender_ID").strip()]
+                })
 
-# Convert cList to a Set to remove duplicate values
-#   -- faster than populating set initially
-# Convert back to list so json.dump can process it
-    arr = set(cList)
-    new_arr = list(arr)
+    df = pd.DataFrame(cList)
+    df.drop_duplicates(subset='Text',inplace = True)
 
-# Write contents of list to json file
-    with open('floop-conv-data.json', 'w') as f:
-        json.dump(new_arr, f)
+# Write contents of document to csv
+    df.to_csv('floop_conv_data.csv',index=False)
```

Comment:
Instead of a csv, we did want json.  Can adjust this easily to json with the following change:

`df.to_json('floop_conv_data.json', orient="records")`